File: examples/import/nonmem/01_advan1_onecomp/convert.jl
# NONMEM ADVAN1 Import - Julia Example
# Run: julia --project=packages/core convert.jl

using Pkg
Pkg.activate("packages/core")

using NeoPKPD
using JSON

println("NONMEM ADVAN1 Import")
println("="^50)

# Import NONMEM control file with dose
model = import_nonmem(
    joinpath(@__DIR__, "run001.ctl");
    doses=[DoseEvent(0.0, 100.0)]
)

println("\nImported Model:")
println("  Type:       $(model.model_type)")
println("  Parameters: $(model.parameters)")
println("  Source:     $(model.source)")

# Display NeoPKPD specification
println("\nNeoPKPD Specification:")
println("-"^50)
spec = model.spec

println("  Model:   $(typeof(spec.kind))")
println("  Params:  $(spec.params)")

if model.iiv !== nothing
    println("  IIV:")
    println("    Kind:   $(model.iiv.kind)")
    println("    Omegas: $(model.iiv.omegas)")
end

if model.error !== nothing
    println("  Error:   $(model.error)")
end

if !isempty(model.warnings)
    println("\n  Warnings:")
    for w in model.warnings
        println("    - $w")
    end
end

# Simulate to verify
println("\n" * "="^50)
println("SIMULATION TEST")
println("="^50)

grid = SimGrid(0.0, 24.0, collect(0.0:1.0:24.0))
solver = SolverSpec(:Tsit5, 1e-10, 1e-12, 10^7)
result = simulate(spec, grid, solver)

println("\nSimulation successful: $(length(result.times)) time points")
println("Cmax: $(round(maximum(result.observations[:conc]), digits=3)) mg/L")

# Save results
output = Dict(
    "model_type" => String(model.model_type),
    "parameters" => Dict(String(k) => v for (k, v) in model.parameters),
    "source" => model.source,
    "cmax" => maximum(result.observations[:conc])
)
open(joinpath(@__DIR__, "output.json"), "w") do io
    JSON.print(io, output, 2)
end
println("\nWrote output.json")

File: examples/import/nonmem/01_advan1_onecomp/convert.py
#!/usr/bin/env python3
"""
NONMEM ADVAN1 Import - Python Example

Run: python convert.py
"""

from openpkpd import import_nonmem, simulate
import json


def main():
    print("NONMEM ADVAN1 Import")
    print("=" * 50)

    # Import NONMEM control file
    model = import_nonmem("run001.ctl")

    print(f"\nImported Model:")
    print(f"  Type:       {model.model_type}")
    print(f"  Parameters: {model.parameters}")

    # Display OpenPKPD specification
    print("\nOpenPKPD Specification:")
    print("-" * 50)
    spec = model.spec

    print(f"  Model:   {spec.model}")
    print(f"  Params:  {spec.params}")

    if spec.iiv:
        print(f"  IIV:")
        print(f"    Kind:   {spec.iiv.kind}")
        print(f"    Omegas: {spec.iiv.omegas}")

    if spec.error:
        print(f"  Error:   {spec.error}")

    # Compare with expected
    print("\n" + "=" * 50)
    print("VALIDATION")
    print("=" * 50)

    with open("expected.json") as f:
        expected = json.load(f)

    # Validate parameters
    params_match = spec.params == expected["params"]
    print(f"Parameters match: {params_match}")

    # Validate IIV
    iiv_match = spec.iiv.omegas == expected["iiv"]["omegas"]
    print(f"IIV match: {iiv_match}")

    # Simulate to verify
    print("\n" + "=" * 50)
    print("SIMULATION TEST")
    print("=" * 50)

    result = simulate(
        spec,
        t_end=24.0,
        saveat=1.0,
        doses=[{"time": 0.0, "amount": 100.0}]
    )

    print(f"\nSimulation successful: {len(result['times'])} time points")
    print(f"Cmax: {max(result['observations']['conc']):.3f} mg/L")

    return model


if __name__ == "__main__":
    main()
